ReactNativeClient/lib/services/searchengine/filterParser.ts

```diff
@@ -0,0 +1,101 @@
+interface Filter {
+	relation: string,
+	name: string,
+	value: string
+}
+
+interface Term {
+  relation: string;
+  value: string;
+}
+
+const parseQuery = (query: string): Array<Filter> => {
+	const terms = [];
+
+	// tag:123 tag:234 or some query -> [["tag", "123"], ["tag", "234"], ["_", "or"], ["_", "some"], ["_", "query"]]
+	let inQuote = false;
+	let currentCol = '_';
+	let currentTerm = '';
+	for (let i = 0; i < query.length; i++) {
+		const c = query[i];
+
+		if (c === '"') {
+			currentTerm += c; // keep the quotes
+			if (inQuote) {
+				terms.push([currentCol, currentTerm]);
+				currentTerm = '';
+				inQuote = false;
+			} else {
+				inQuote = true;
+			}
+			continue;
+		}
+
+		if (c === ' ' && !inQuote) {
+			if (!currentTerm) continue;
+			terms.push([currentCol, currentTerm]);
+			currentCol = '_';
+			currentTerm = '';
+			continue;
+		}
+
+		if (c === ':' && !inQuote) {
+			currentCol = currentTerm;
+			currentTerm = '';
+			continue;
+		}
+
+		currentTerm += c;
+	}
+	if (currentTerm) terms.push([currentCol, currentTerm]);
+
+
+	let relation = 'AND'; // default relation
+	const result: Array<Filter> = [];
```

Comment:
To define array please use the syntax `const result: Filter[] = []` (In general we should avoid using the Array class directly)